File: src/main/java/me/nylestroke/nylemod/entity/client/item/GoldenStandItemModel.java
package me.nylestroke.nylemod.entity.client.item;

import me.nylestroke.nylemod.NylemodExample;
import me.nylestroke.nylemod.item.custom.GoldenStandItem;
import net.minecraft.util.Identifier;
import software.bernie.geckolib3.model.AnimatedGeoModel;

public class GoldenStandItemModel extends AnimatedGeoModel<GoldenStandItem> {

    @Override
    public Identifier getModelLocation(GoldenStandItem object) {
        return new Identifier(NylemodExample.MOD_ID, "geo/golden_stand.geo.json");
    }

    @Override
    public Identifier getTextureLocation(GoldenStandItem object) {
        return new Identifier(NylemodExample.MOD_ID, "textures/machines/golden_stand.png");
    }

    @Override
    public Identifier getAnimationFileLocation(GoldenStandItem animatable) {
        return new Identifier(NylemodExample.MOD_ID, "animations/golden_stand.animation.json");
    }
    
}

File: src/main/kotlin/me/nylestroke/nylemod/item/entity/client/GoldenStandItemModel.kt
package me.nylestroke.nylemod.item.entity.client

import me.nylestroke.nylemod.NylemodExample
import me.nylestroke.nylemod.item.custom.GoldenStandItem
import net.minecraft.util.Identifier
import software.bernie.geckolib3.model.AnimatedGeoModel

class GoldenStandItemModel : AnimatedGeoModel<GoldenStandItem?>() {
    override fun getModelResource(`object`: GoldenStandItem?): Identifier {
        return Identifier(NylemodExample.MOD_ID, "geo/golden_stand.geo.json")
    }

    override fun getTextureResource(`object`: GoldenStandItem?): Identifier {
        return Identifier(NylemodExample.MOD_ID, "textures/machines/golden_stand.png")
    }

    override fun getAnimationResource(animatable: GoldenStandItem?): Identifier {
        return Identifier(NylemodExample.MOD_ID, "animations/golden_stand.animation.json")
    }
}